doc/hooks/versionCheckHook.section.md

```diff
@@ -16,20 +22,67 @@ stdenv.mkDerivation (finalAttrs: {
     versionCheckHook
   ];
+
   doInstallCheck = true;
 
   # ...
 })
 ```
 
-Note that for [`buildPythonPackage`](#buildpythonpackage-function) and [`buildPythonApplication`](#buildpythonapplication-function), `doInstallCheck` is enabled by default.
+:::
+
+Note that `doInstallCheck` is enabled by default for [`buildPythonPackage`](#buildpythonpackage-function) and [`buildPythonApplication`](#buildpythonapplication-function).
+
+## Attributes controlling versionCheckHook {#versioncheckhook-attributes-controlling}
+
+The following attributes control `versionCheckHook`:
+
+- `dontVersionCheck`: When set as `true`, the hook is not added to the [`preInstallCheckHooks`](#ssec-installCheck-phase).
+
+  Useful when one wants to load the function defined by the hook, so that it can be used for a different purpose.
+
+  Defaults to false.
+
+- `versionCheckProgram`: The full path to the program to be executed.
+
+   Defaults to `${placeholder "out"}/bin/${pname}`, roughly speaking.
+
+   The usage of `placeholder "out"` instead of `$out` here is unavoidable, since the hook does not expand environment variables.
+
+- `versionCheckProgramArg`: The argument to be passed to `versionCheckProgram`.
+
+  When not defined, the hook tries `--help` first and then `--version`.
+
+  Examples: `version`, `-V`, `-v`.
+
+- `preVersionCheck`: Hook that runs before the check is done. Defaults to an empty string.
+
+- `postVersionCheck`: Hook that runs after the check is done. Defaults to an empty string.
+
+Note: see also [`testers.testVersion`](#tester-testVersion).
+
+## Comparison between `versionCheckHook` and `testers.testVersion` {#versioncheckhook-comparison-testversion}
+
+The most notable difference between `versionCheckHook` and `testers.testVersion` is that `versionCheckHook` runs as a phase during the build time of the package, while `passthru.tests.versions` executes a whole derivation that depends on the package.
+
+Given the current limitations of Nixpkgs's tooling and CI, `versionCheckHook` fits better in most of typical situations.
+
+Below we tabulate the differences between `versionCheckHook` and `testers.testVersion`.
+
+
+| Item                                   | `versionCheckHook`                                                                                                                                            | `testers.testVersion`                                                                                                                                                                                       |
+|:---------------------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------:|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------:|
```

Comment:
Nit: Looks better IMO when lines are aligned to the left.

Reply:
I will remove these lines, at least until I discover why the manual was not built.